matrix/tests_matrix.py

```diff
@@ -0,0 +1,190 @@
+import unittest
+
+from matrix import Matrix
+
+
+class TestMatrix(unittest.TestCase):
+    def _test_initialization(self, *data):
+        try:
+            Matrix(*data)
+        except ValueError:
+            self.fail("Initialization raised ValueError")
```

Comment:
Exception raised unexpectedly will fail the test so you don't have to `try-expect` it